spec.md

```diff
@@ -286,6 +286,67 @@ circuit Foo :
   public module Foo enablelayer A :
 ```
 
+## Formal Tests
+
+Formal tests are named top-level constructs which allow for formal test harnesses to be definied 
+as bounded model checking (BMC) problems. The body of the formal test harness is converted 
+into a BMC formula which encodes a state-transition system version of the design using one 
+of the formal backends (targetting either BTOR2 or SMTLib). 
+Formal tests have 3 operands: 
+- A symbol that the test can be referred to with.  
+- The name of the formal test harness module.  
+- A bound for the bounded model checking problem, which refers to the maximum bound k used
+to check the bounded model checking formula. This means that given a bound k, the BMC formula 
+will be checked for all unrollings of the resulting state-transition system of depths 1 through k.
+This bound is a strictly positive integer.  
+
+```firrtl
+FIRRTL version 4.0.0
+circuit Foo :
+  public module Foo :
+    ; ...
+  formal testFoo of Foo, bound = 20
+```
+More details about how bounded model checking works and what the bound refers to can be found in 
+[Biere et al. 2003](https://www.cs.cmu.edu/~emc/papers/Books%20and%20Edited%20Volumes/Bounded%20Model%20Checking.pdf).
+ 
+### Formal Test Harness  
+
+A formal test harness is a public module that is typically only used for defining the 
+body of a formal test-bench. These are modules that are not intended to be instantiated and 
+their inputs function as free (or symbolic) variables. Formal test harnesses typically 
```

Comment:
With this sentence and the one above, you use the word "typically".

We should be careful when we use this word. It conveys a norm rather than a rule.

Reply:
I just wanted to clarify that this isn't something that was made-up for firrtl and that it's rather a widely used concept already that can be used however you like but is expected to be used in a certain way -- not sure if that makes sense, I can remove the word typically to avoid any confusion.